### PR TITLE
Mdbenjam review to react

### PR DESCRIPTION
**Pull request by mdbenjam**
Making the review page in React to make the form persist even when flipping back to the review page.

**Test Plan**
- [ ] Test manually
- [ ] Update tests for the new flow
- [ ] Run test suite
reactpage

### CODE REVIEW

File: spec/feature/establish_claim_spec.rb [hunk 1]
```diff
@@ -115,10 +116,12 @@
     scenario "Return to decision on an Establish Claim task returns me decision review" do
       @task.assign!(current_user)
       visit "/dispatch/establish-claim/#{@task.id}/new"
+      click_on "Create End Product"
       expect(page).to have_content("Create End Product") # React works
 
       click_on "\u00ABBack to review"
-      expect(page).to have_current_path("/dispatch/establish-claim/#{@task.id}/review")
+      expect(page).to have_current_path("/dispatch/establish-claim/#{@task.id}/new")
```
Comment (shanear) [inline spec/feature/establish_claim_spec.rb]: would it be worth it to add a test to see that values entered stick around?

File: client/app/containers/EstablishClaim.jsx [hunk 2]
```diff
@@ -215,22 +226,76 @@ export default class EstablishClaim extends React.Component {
...
+      this.handleSubmit(event);
+    } else {
+      throw "Invalid page state";
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        { this.state.page == REVIEW_PAGE && this.review() }
```
Comment (joofsh) [inline client/app/containers/EstablishClaim.jsx]: `===` in javascript please. `==` means something else. Does linting not catch this? We should add it

File: client/app/containers/EstablishClaim.jsx [hunk 3]
```diff
@@ -225,24 +235,104 @@ export default class EstablishClaim extends React.Component {
...
+            <a href="#send_to_ro" className="cf-btn-link cf-adjacent-buttons">
+              Send to RO
+            </a>
+            <Button
+              name="Create End Product"
+              loading={loading}
+              onClick={this.handleCreateEndProduct}
+            />
+          </div>
+          { this.state.page === FORM_PAGE &&
```
Comment (joofsh) [inline client/app/containers/EstablishClaim.jsx]: you have a bunch of these `page === REVIEW_PAGE` on line 304, and then this one on 318. Can we abstract these to helper methods?

`this.isReviewPage()` and `this.isFormPage()`

File: app/controllers/tasks_controller.rb [hunk 4]
```diff
@@ -12,6 +12,13 @@ def index
     render index_template
   end
 
+  def new
+    # Future safeguard for when we give managers a show view
+    # for a given task
+    task.start! if current_user == task.user
+    render "new"
```
Comment (joofsh) [inline app/controllers/tasks_controller.rb]: this shouldn't be necessary, will automatically render `new.html.erb`

**Pull request merged**